crates/ra_fmt2/src/indent.rs

```diff
@@ -119,25 +119,33 @@ impl<'b> Default for IndentBuilder<'b> {
 
 impl<'b> IndentBuilder<'b> {
```

Comment:
`IndentBuilder` holds the walked blocks calculating indent so nesting will eventually work hopefully. How do i get a `IndentRule` from a `PatternSet` there is no way to `AsRef` it because of the options right? `Indentation` lives on `Block` it seemed easier to have more direct access to the block instead of having to go through `Whitespace`, is this ok?

Reply:
Hm, I think rather than calculcating current indent as we go, we should re-compute indent for each node.

That is, to indent node `n`, you walk it's ancestors until you see an "anchoring" node, relative to which `n` should be indented. Then, you check current indent of the anchor and indent `n` relative to that. It should help with cases where one wants to re-indent a specific node, without touching anything outside.